src/tribler/gui/event_request_manager.py

```diff
@@ -37,9 +37,10 @@ class EventRequestManager(QNetworkAccessManager):
 
     def __init__(self, api_port, api_key, error_handler):
         QNetworkAccessManager.__init__(self)
-        url = QUrl("http://localhost:%d/events" % api_port)
-        self.request = QNetworkRequest(url)
-        self.request.setRawHeader(b'X-Api-Key', api_key.encode('ascii'))
+        self.api_port = api_port
+        self.api_key = api_key
+        self.request = None
```

Comment:
Nitpick: it is possible to create a request right now and skip the `if self.request is None` check later:
```python
        self.request = self.create_request()
```